src/otg/dataset/study_index.py

```diff
@@ -477,3 +477,113 @@ def from_source(
                 )
             )
         )
+
+
+@dataclass
+class StudyIndexUKBiobank(StudyIndex):
+    """Study index dataset from UKBiobank.
+
+    The following information is extracted:
+
+    - studyId
+    - pubmedId
+    - publicationDate
+    - publicationJournal
+    - publicationTitle
+    - publicationFirstAuthor
+    - traitFromSource
+    - ancestry_discoverySamples
+    - ancestry_replicationSamples
+    - initialSampleSize
+    - nCases
+    - replicationSamples
+
+    Some fields are populated as constants, such as projectID, studyType, and initial sample size.
+    """
+
+    @classmethod
+    def from_source(
+        cls: type[StudyIndexUKBiobank],
+        ukbiobank_studies: DataFrame,
+    ) -> StudyIndexUKBiobank:
+        """This function ingests study level metadata from UKBiobank.
+
+        Args:
+            ukbiobank_studies (DataFrame): UKBiobank study manifest file loaded in spark session.
+
+        Returns:
+            StudyIndexUKBiobank: Annotated UKBiobank study table.
+        """
+        return cls(
+            _df=(
+                ukbiobank_studies.select(
+                    f.col("code").alias("studyId"),
+                    f.col("trait").alias("traitFromSource"),
+                    f.col("n_cases").cast("long").alias("nCases"),
+                    f.col("n_total").cast("string").alias("initialSampleSize"),
+                    f.col("in_path").alias("summarystatsLocation"),
+                )
+                # Replace any trait double spaces with single.
+                .withColumn(
+                    "traitFromSource",
+                    f.regexp_replace(f.col("traitFromSource"), r" +", " "),
+                )
+                # Swap trait prefix and suffix positions.
+                .withColumn(
+                    "traitFromSource",
+                    f.when(
+                        f.col("traitFromSource").contains(":"),
+                        f.concat(
+                            f.initcap(
+                                f.split(f.col("traitFromSource"), ": ").getItem(1)
+                            ),
+                            f.lit(" | "),
+                            f.lower(f.split(f.col("traitFromSource"), ": ").getItem(0)),
+                        ),
+                    ).otherwise(f.col("traitFromSource")),
+                )
+                # Make publication and ancestry schema columns.
+                .withColumn(
+                    "publicationDate",
+                    f.when(
+                        f.col("studyId").startswith("NEALE2_"),
+                        "2018-08-01",
+                    ).otherwise("2018-10-24"),
+                )
+                .withColumn("pubmedId", f.lit(""))
+                .withColumn("publicationJournal", f.lit(""))
+                .withColumn("publicationTitle", f.lit(""))
```

Comment:
I assume the reason we're leaving those fields blank is that the UK Biobank studies don't have an associated literature information?

@d0choa Can you please remind me what do we want to do in this case? Just leave them blank? Or perhaps hardcode one citation for the entire UK Biobank?

(Although I just discovered that they don't have a paper on that for some reason, and are just asking to cite... the URL: http://www.nealelab.is/uk-biobank/faq)

Reply:
NEALEs' certainly don't have a publication, but we can referece SAIGE with [30104761](https://pubmed.ncbi.nlm.nih.gov/30104761/)

Instead of empty strings, I would leave these fields as null.

Reply:
Thank you for this, just to be sure, should I add in the SAIGE reference and leave all other fields as null, something like:
```
.withColumn("pubmedId",
                    f.when(
                        f.col("studyId").startswith("SAIGE_"),
                        "30104761",
                    ).otherwise(None),
                )

.withColumn("publicationJournal",
                    f.when(
                        f.col("studyId").startswith("SAIGE_"),
                        "Nature Genetics",
                    ).otherwise(None),
                )

.withColumn("publicationTitle",
                    f.when(
                        f.col("studyId").startswith("SAIGE_"),
                        "Efficiently controlling for case-control imbalance and sample relatedness in large-scale genetic association studies",
                    ).otherwise(None),
                )
```
Or instead just leave all as null with:
```
.withColumn("pubmedId", f.lit(None))
.withColumn("publicationJournal", f.lit(None))
.withColumn("publicationTitle", f.lit(None))
```